src/core/thread/network_data.cpp

```diff
@@ -977,7 +977,9 @@ otError NetworkData::SendServerDataNotification(uint16_t aRloc16)
     Coap::Message *  message = NULL;
     Ip6::MessageInfo messageInfo;
 
-    VerifyOrExit(mLastAttempt.GetValue() == 0 || ((TimerMilli::GetNow() - mLastAttempt) > kDataResubmitDelay),
+    VerifyOrExit(mLastAttempt.GetValue() == 0 ||
+                     ((mType == kTypeLocal) && (TimerMilli::GetNow() - mLastAttempt) > kDataResubmitDelay) ||
```

Comment:
Use local variable to store `TimerMilli::GetNow() - mLastAttempt` to avoid computing twice?